core/templates/dev/head/pages/library/Library.js

```diff
@@ -80,6 +80,31 @@ oppia.controller('Library', [
       $http.get('/libraryindexhandler').success(function(data) {
         $scope.libraryGroups = data.activity_summary_dicts_by_category;
 
+        $http.get('/creatordashboardhandler/data')
+          .then(function(response) {
+            if (response.data.explorations_list !== undefined) {
+              for (var i = 0; i < $scope.libraryGroups.length; i++) {
+                for (var j = 0; j < $scope.libraryGroups[i]
+                  .activity_summary_dicts.length; j++) {
+                  var activity = $scope.libraryGroups[i].activity_summary_dicts;
+                  activity[j].should_add_to_playlist_icon = true;
+                  for (var k = 0; k < response.data
+                    .explorations_list.length; k++) {
+                    if (response.data.explorations_list.id !== activity[j].id) {
+                      activity[j].should_add_to_playlist_icon = false;
```

Comment:
For maintainability, it is not OK to modify a dict dynamically. This results in uncertainty over which keys the dict contains, since the keyset changes depending on where you are in the code.

Instead, I suggest you take the response value from the backend and parse it into a dictionary that looks something like {explorations: {expid1: true, expid2: true, ...}, collections: {colid1: true, colid2: true, ...}}. Then you can get the "should show add to playlist icon" boolean value using e.g. `!!d[explorations][expid1]`.

Reply:
understood.